### PR TITLE
Xiana release 0.5.0

### PR DESCRIPTION
**Pull request by g-krisztian**

### CODE REVIEW

**Review by soulflyer — COMMENTED** [review]
I don't think docs/0.3.0 files should be changed at all. Better to run codox and generate new docs for 0.4.0 I think?

**Review by soulflyer — APPROVED** [review]

**Comment by mihainem** [comment]
All tests passed ok for me

**Review by mihainem — APPROVED** [review]

File: examples/acl/project.clj [hunk 1]
```diff
@@ -1,8 +1,7 @@
 (defproject acl "0.1.0-SNAPSHOT"
   :description "FIXME: write description"
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/clojure "1.10.3"]
-                 [com.flexiana/framework "0.4.0-rc5"]]
+  :dependencies [[com.flexiana/framework "0.5.0-rc1"]]
```
Comment (petervlk) [inline examples/acl/project.clj:4]: why is clojure removed from the dependencies? we are directly using it in the project, but this way clojure is only transitive dependency brought in by framework. In my opinion, this should be avoided.

File: examples/cli-chat/project.clj [hunk 2]
```diff
@@ -1,8 +1,7 @@
 (defproject cli-chat "0.1.0-SNAPSHOT"
   :description "FIXME: write description"
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/clojure "1.10.3"]
-                 [com.flexiana/framework "0.4.0-rc5"]]
+  :dependencies [[com.flexiana/framework "0.5.0-rc1"]]
```
Comment (petervlk) [inline examples/cli-chat/project.clj:4]: same transitive dependency on clojure as described previously

File: examples/controllers/project.clj [hunk 3]
```diff
@@ -1,9 +1,8 @@
 (defproject controllers "0.1.0-SNAPSHOT"
   :description "FIXME: write description"
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/clojure "1.10.3"]
-                 [org.clojure/clojurescript "1.11.4"]
-                 [com.flexiana/framework "0.4.0-rc5"]]
+  :dependencies [[com.flexiana/framework "0.5.0-rc1"]
```
Comment (petervlk) [inline examples/controllers/project.clj:4]: same transitive dependency on clojure as described previously

File: examples/frames/project.clj [hunk 4]
```diff
@@ -1,8 +1,7 @@
 (defproject frames "0.1.0-SNAPSHOT"
   :description "FIXME: write description"
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/clojure "1.10.3"]
-                 [com.flexiana/framework "0.4.0-rc5"]]
+  :dependencies [[com.flexiana/framework "0.5.0-rc1"]]
```
Comment (petervlk) [inline examples/frames/project.clj:4]: same transitive dependency on clojure as described previously

File: examples/sessions/project.clj [hunk 5]
```diff
@@ -1,25 +1,26 @@
 (defproject sessions "0.1.0-SNAPSHOT"
   :description "FIXME: write description"
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/clojure "1.10.3"]
-                 [com.flexiana/framework "0.4.0-rc5"]]
-  :plugins [[lein-shadow "0.3.1"]
+  :dependencies [[com.flexiana/framework "0.5.0-rc1"]]
```
Comment (petervlk) [inline examples/sessions/project.clj:4]: same transitive dependency on clojure as described previously

File: examples/state-events/project.clj [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
   :description "FIXME: write description"
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/clojure "1.10.3"]
-                 [org.clojure/clojurescript "1.11.4"]
-                 [com.flexiana/framework "0.4.0-rc5"]
-                 [thheller/shadow-cljs "2.17.7"]
-                 [re-frame "1.2.0"]
-                 [cljs-ajax "0.8.4"]
-                 [org.clojure/tools.namespace "1.2.0"]
+  :dependencies [[com.flexiana/framework "0.5.0-rc1"]
```
Comment (petervlk) [inline examples/state-events/project.clj:4]: same transitive dependency on clojure as described previously

**Review by petervlk — COMMENTED** [review]
looks good, gut I dont like the transitive dependencies on clojure in examples. in my opinion the stuff that we directly use in our projects should be directly declared (transparent dependencies).

File: examples/acl/project.clj (hunk 1 above)
Comment (jehaby) [reply examples/acl/project.clj:4]: I agree that usually it's preferable to have explicit dependency on clojure in a project. The main reasoning behind this was avoiding repetition (necessity to change clojure version in all examples each time we update it in the framework). Considering it works I would prefer to leave it as is.

Only downsides of relying on transitive dependency I see atm is the possibility of the dependency being removed from an intermediate project. But this should be pretty easily fixable on the compilation stage. And it's not like we have plans to remove clojure deps from framework.
If you have any other arguments pls let me know.

**Review by petervlk — APPROVED** [review]
LGTM :+1:

**Pull request merged**